apps/api/src/auth/services/api-key/api-key-auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import isAfter from "date-fns/isAfter";
+import parseISO from "date-fns/parseISO";
+import { singleton } from "tsyringe";
+
+import { ApiKeyOutput, ApiKeyRepository } from "@src/auth/repositories/api-key/api-key.repository";
+import { ApiKeyGeneratorService } from "./api-key-generator.service";
+
+@singleton()
+export class ApiKeyAuthService {
+  constructor(
+    private readonly apiKeyGenerator: ApiKeyGeneratorService,
+    private readonly apiKeyRepository: ApiKeyRepository
+  ) {}
+
+  async validateApiKeyFromHeader(apiKey: string | undefined): Promise<ApiKeyOutput | undefined> {
```

Comment:
if it's validating should it throw?

Reply:
@ygrishajev I personally would use exceptions only for exceptional situations because exceptions type cannot be injected in function declaration. But we can incorporate it into return type and by doing this we force caller to process the result and do explicit decision regarding error handling. Additionally it serves documentation purposes

Reply:
go "go"